### PR TITLE
chore: move to tekton-ci

### PR DESCRIPTION
**Pull request by psturc**
https://issues.redhat.com/browse/KONFLUX-2245

* related changes [already merged to infra-deployments](https://github.com/redhat-appstudio/infra-deployments/pull/3643)
* app is running on a staging cluster and is already being used

### CODE REVIEW

File: .tekton/ci-helper-app-pull-request.yaml [hunk 1]
```diff
@@ -0,0 +1,38 @@
+apiVersion: tekton.dev/v1beta1
+kind: PipelineRun
+metadata:
+  name: ci-helper-app-pull-request
+  annotations:
+    pipelinesascode.tekton.dev/on-event: "[pull_request]"
+    pipelinesascode.tekton.dev/on-target-branch: "[main]"
+    pipelinesascode.tekton.dev/max-keep-runs: "5"
```
Comment (dheerajodha) [inline .tekton/ci-helper-app-pull-request.yaml:8]: under which namespace would this PLR be running?

File: deploy/base/kustomization.yaml
Comment (dheerajodha) [inline deploy/base/kustomization.yaml:1]: dumb question: why are these deployment-related files here in this repo? seems like they should be in infra-deployments

Comment (psturc) [reply deploy/base/kustomization.yaml:1]: we use to store deployments, routes etc. manifests in component repositories. ([example](https://github.com/konflux-ci/integration-service/tree/main/config/manager))

infra-deployments contains [only the data for overriding the image tag and environment specific stuff like rbac etc.](https://github.com/redhat-appstudio/infra-deployments/tree/main/components/ci-helper-app)

File: .tekton/ci-helper-app-pull-request.yaml (hunk 1 above)
Comment (psturc) [reply .tekton/ci-helper-app-pull-request.yaml:8]: that depends on the namespace where [related repository CR is created](https://github.com/redhat-appstudio/infra-deployments/blob/main/components/konflux-ci/base/repository.yaml). in this case it's `konflux-ci`

**Review by dheerajodha — APPROVED** [review]

**Pull request merged**